### PR TITLE
executor, ddl: get the correct result of "show create table" when adding the first index to the table

### LINKED ISSUE #9386: Get the correct result of "show create table" when adding the first index to the table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

create table t (id int); 
session 1:
```
alter table t add index idx(id);
```
session 2:
```
show create table t;
```
We execute the statement of `show create table t;` when the statement of `alter table t add index idx(id);` is running.

2. What did you expect to see?
```
+-------+--------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                             |
+-------+--------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `id` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?
```
+-------+--------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                             |
+-------+--------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `id` int(11) DEFAULT NULL,

) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
Release Version: v3.0.0-beta-92-gf46c558a6
Git Commit Hash: f46c558a60441e9c79678d1ed946a6cc81765243
Git Branch: master

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Fix #9386
create table t (id int); 
session 1:
```
alter table t add index idx(id);
```
session 2:
```
show create table t;
```
We execute the statement of `show create table t;` when the statement of `alter table t add index idx(id);` is running.

Before this PR:
```
+-------+--------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                             |
+-------+--------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `id` int(11) DEFAULT NULL,

) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
Now:
```
+-------+--------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                             |
+-------+--------------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `id` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: ddl/db_change_test.go [hunk 1]
```diff
@@ -80,9 +80,10 @@ func (s *testStateChangeSuite) TearDownSuite(c *C) {
 func (s *testStateChangeSuite) TestShowCreateTable(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
 	tk.MustExec("use test")
-	tk.MustExec("create table t (id int, index idx (id))")
+	tk.MustExec("create table t (id int)")
 
 	var checkErr error
+	addIndexSeq := 0
```
Comment (shenli) [inline ddl/db_change_test.go]: Could you add a comment for this variable?

Comment (zimulala) [reply ddl/db_change_test.go]: I update the test code.
PTAL @shenli

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**